server/src/main/java/org/apache/iotdb/db/newsync/pipedata/queue/BufferedPipeDataQueue.java

```diff
@@ -0,0 +1,423 @@
+package org.apache.iotdb.db.newsync.pipedata.queue;
+
+import org.apache.iotdb.db.exception.metadata.IllegalPathException;
+import org.apache.iotdb.db.newsync.conf.SyncConstant;
+import org.apache.iotdb.db.newsync.pipedata.PipeData;
+import org.apache.iotdb.db.newsync.pipedata.TsFilePipeData;
+import org.apache.iotdb.db.utils.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.LinkedBlockingDeque;
+
+public class BufferedPipeDataQueue implements PipeDataQueue {
+  private static final Logger logger = LoggerFactory.getLogger(BufferedPipeDataQueue.class);
+
+  private final String pipeLogDir;
+
+  /** input */
+  private long lastMaxSerialNumber;
+  private BlockingDeque<PipeData> inputDeque;
+
+  private BlockingDeque<Long> pipeLogStartNumber;
+  private DataOutputStream outputStream;
+  private long currentPipeLogSize;
+
+  /** output */
+  private final Object waitLock = new Object();
+  private BlockingDeque<PipeData> outputDeque;
+
+  private long pullSerialNumber;
+  private long commitSerialNumber;
+  private BufferedWriter commitLogWriter;
+  private long currentCommitLogSize;
+
+  public BufferedPipeDataQueue(String pipeLogDir) {
```

Comment:
make it private and provide a static funtion to create BufferedPipeDataQueue

Reply:
`BufferedPipeDataQueue` is not a singleton.